sdk/database/dbplugin/v5/grpc_server_test.go

```diff
@@ -253,13 +257,17 @@ func TestGRPCServer_NewUser(t *testing.T) {
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			g := gRPCServer{
-				impl: test.db,
+				factoryFunc: func() (interface{}, error) {
+					return test.db, nil
+				},
+				instances: make(map[string]Database),
 			}
 
-			// Context doesn't need to timeout since this is just passed through
-			ctx := context.Background()
+			id := "12345"
+			idCtx := idCtx(t, id)
+			g.instances[id] = test.db
```

Comment:
This setup logic is being repeated more than once or twice, wondering if it'd be better to factor it out to something like:

`testGrpcServer(t *testing.T, db Database) (context.Context, gRPCServer)`

Reply:
Good idea!